rest-service/manager_rest/rest/resources_v3_1/execution_schedules.py

```diff
@@ -0,0 +1,275 @@
+        else:
+            rule = schedule.rule.copy()
+            if 'rrule' in rule:
+                del rule['rrule']  # RRULE will be calculated anew later
+            if count:
+                rule['count'] = convert_to_int(count)
+            if frequency:
+                rule['frequency'] = self._verify_frequency(frequency)
+            if weekdays:
+                rule['weekdays'] = self._verify_weekdays(weekdays)
+            schedule.rule = rule
+
+        sm.update(schedule)
+        return schedule, 201
+
+    @swagger.operation(
+        responseClass=models.ExecutionSchedule,
+        nickname="getById",
+        notes="Returns the execution schedule by its id.",
+    )
+    @authorize('execution_get')
+    @rest_decorators.marshal_with(models.ExecutionSchedule)
+    def get(self, schedule_id, _include=None, **kwargs):
+        """
+        Get execution schedule by id
+        """
+        # TODO :: include in response: number of executions already run,
+        #         of them - number succeeded + number failed
+
+        return get_storage_manager().get(
+            models.ExecutionSchedule,
+            schedule_id,
+            include=_include
+        )
+
+    @authorize('execution_delete')
+    def delete(self, schedule_id):
+        sm = get_storage_manager()
+        schedule = sm.get(models.ExecutionSchedule, schedule_id)
+        sm.delete(schedule)
+        return None, 204
+
+    @staticmethod
+    def _verify_timestamp(date_str):
+        if not date_str:
+            return
+        for fmt in ('%Y-%m-%d', '%Y-%m-%dT%H:%M:%S',
+                    '%Y-%m-%d %H:%M:%S', '%Y-%m-%d %H:%M:%S.%f',
+                    '%Y-%m-%dT%H:%M:%S.%f', '%Y-%m-%dT%H:%M:%S.%fZ'):
+            try:
+                datetime.strptime(date_str, fmt)
+                return date_str
+            except ValueError:
+                pass
+        raise manager_exceptions.BadParametersError(
+            "{} is not a legal UNIX time".format(date_str))
+
+    @staticmethod
+    def _verify_frequency(frequency_str):
+        if not frequency_str:
+            return
+        parsed = re.findall(r"(\d+) (seconds?|minutes?|hours?|days?|weeks?"
+                            "|months?|years?) ?(ago)?", frequency_str)
+        if not parsed or len(parsed[0]) < 2:
+            raise manager_exceptions.BadParametersError(
+                "`{}` is not a legal frequency expression. Supported format "
+                "is: <number> seconds|minutes|hours|days|weeks|months|years"
+                .format(frequency_str))
+        return frequency_str
+
+    @staticmethod
+    def _verify_weekdays(weekdays):
+        if not weekdays:
+            return
+        if not isinstance(weekdays, list):
+            raise manager_exceptions.BadParametersError(
+                "weekdays should be a list, got {}".format(type(weekdays)))
+        weekdays_caps = set(d.upper() for d in weekdays)
+        valid_weekdays = [str(d) for d in rrule.weekdays]
```

Comment:
no reason for this to be a list, a set would be better
then you could also `invalid = weekdays_caps - valid_weekdays` and skip the forloop